### PR TITLE
Fix: connection should not auto retry after DirectLineStreaming.end() 

### PR DESCRIPTION
**Pull request by woasishen**
Check wether it is manual end connection when retry to connect.

### CODE REVIEW

**Review by compulim — APPROVED** [review]

**Review by compulim — COMMENTED** [review]
Let's add some tests.

**Review by compulim — CHANGES_REQUESTED** [review]
Let's add a test.

File: __tests__/happy.dlstreamConnection.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import waitForConnected from './setup/waitForConnected';
+
+describe('test dl streaming end', () => {
+  let unsubscribes;
+  let directLine;
+  const ConnectionStatusEnd = 5;
+  beforeEach(() => unsubscribes = []);
+  afterEach(() => unsubscribes.forEach(fn => onErrorResumeNext(fn)));
+  test('using Streaming Extensions', async () => {
+    jest.setTimeout(30000);
```
Comment (compulim) [inline __tests__/happy.dlstreamConnection.js:16]: I think we need to expand this test a bit to make sure "when connection status ended, it should not retry."

I am not blocking the PR. Please add the following comment:

```js
// TODO: [hawo] Need more realistic testing.
//       - Able to connect to a Web Socket server
//       - Make sure after `end` is called, the client will not reconnect
//       - If the connection is disrupted, make sure the client will reconnect
```

**Review by compulim — DISMISSED** [review]
Not blocking, please add TODO comment.

**Review by luhan2017 — APPROVED** [review]

**Review by tracyboehrer — APPROVED** [review]

**Pull request merged**